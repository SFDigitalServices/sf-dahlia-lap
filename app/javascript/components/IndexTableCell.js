import React from 'react'
import _ from 'lodash'

const IndexTableCell = ({ attrs, val, editVal, editing, onChange }) => {
  if (editing && attrs && attrs.editable) {
    if (attrs.editable_options) {
      const options = []
      let i = 0
      _.each(attrs.editable_options, (option) => {
        options.push(
          <option value={option} key={i++}>{option}</option>
        )
      })
      return (
        <select value={editVal} onChange={onChange}>
          {options}
        </select>
      )
    } else {
      return (
        <div>
          <input type='text' value={editVal} onChange={onChange} />
        </div>
      )
    }
  } else {
    // Avoid rendering issues by converting undefined to null.
    return val === undefined ? null : val
  }
}

export default IndexTableCell
