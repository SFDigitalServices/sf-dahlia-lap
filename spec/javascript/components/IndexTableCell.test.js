import React from 'react'

import renderer from 'react-test-renderer'

import IndexTableCell from 'components/IndexTableCell'

// TODO: Add tests for editable cell functionality.
describe('IndexTableCell', () => {
  test('should return null if val is undefined', () => {
    const val = null
    const component = renderer.create(<IndexTableCell {...{ val }} />)

    expect(component.toJSON()).toBeNull()
  })

  test('should return string value if cell is not being edited', () => {
    const val = 'test'
    const editing = false
    const component = renderer.create(<IndexTableCell {...{ val, editing }} />)

    expect(component.toJSON()).toEqual(val)
  })
})
