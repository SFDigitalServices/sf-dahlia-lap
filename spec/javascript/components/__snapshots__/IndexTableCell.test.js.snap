// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexTableCell should return null if val is undefined 1`] = `<DocumentFragment />`;

exports[`IndexTableCell should return string value if cell is not being edited 1`] = `
<DocumentFragment>
  test
</DocumentFragment>
`;
